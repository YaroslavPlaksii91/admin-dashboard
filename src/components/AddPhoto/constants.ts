export const IMAGE_SIZE = Object.freeze({
  SMALL: 44,
  BIG: 60,
});
