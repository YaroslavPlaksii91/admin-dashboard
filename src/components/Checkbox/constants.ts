export const CHECKBOX_TEST_ID = 'checkbox';
