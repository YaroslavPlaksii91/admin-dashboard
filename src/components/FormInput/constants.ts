export const TOGGLE_BUTTON_TEST_ID = 'toggleButton';
