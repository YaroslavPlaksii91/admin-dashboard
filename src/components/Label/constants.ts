export const LABEL_TYPES = Object.freeze({
  ROUND: 'round',
  SQUARE: 'square',
});

export const LABEL_COLORS = Object.freeze({
  DEFAULT: 'default',
  SUCCESS: 'success',
  WARNING: 'warning',
  ERROR: 'error',
});
