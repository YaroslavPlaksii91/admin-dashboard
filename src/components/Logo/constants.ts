export const LOGO_TYPES = Object.freeze({
  VERTICAL: 'vertical',
  HORIZONTAL: 'horizontal',
});
