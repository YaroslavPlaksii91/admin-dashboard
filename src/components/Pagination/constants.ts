export const ROWS_PER_PAGE = 8;
export const ROWS_PER_PAGE_OPTIONS = [8, 12, 16, 20];
