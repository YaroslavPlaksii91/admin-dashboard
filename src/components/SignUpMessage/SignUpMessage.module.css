.wrapper {
  display: flex;
  gap: 5px;
  justify-content: center;
}

.text {
  font-size: 14px;
  line-height: 1.43;
  color: var(--secondary-text-color);
}
