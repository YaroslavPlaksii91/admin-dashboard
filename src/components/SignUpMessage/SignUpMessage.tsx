import { FC } from 'react';

import { ROUTES } from '../../routes/constants';
import { Link } from '..';

import styles from './SignUpMessage.module.css';

export const SignUpMessage: FC = () => {
  return (
    <div className={styles.wrapper}>
      <p className={styles.text}>Don't have an account?</p>
      <Link text="Sign up" url={ROUTES.REGISTER_PAGE} />
    </div>
  );
};
