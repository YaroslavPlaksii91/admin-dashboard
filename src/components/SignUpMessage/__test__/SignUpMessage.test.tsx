import { MemoryRouter } from 'react-router-dom';
import { cleanup, render, screen } from '@testing-library/react';

import { ROUTES } from '@routes/constants';
import { SignUpMessage } from '../SignUpMessage';

describe('SignUpMessage component', () => {
  afterEach(cleanup);

  it('should be rendered', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <SignUpMessage />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should have correct text', () => {
    const text = "Don't have an account?";

    render(
      <MemoryRouter>
        <SignUpMessage />
      </MemoryRouter>,
    );

    const message = screen.getByText(text);

    expect(message).toBeInTheDocument();
  });

  it('should contain link to register page', () => {
    render(
      <MemoryRouter>
        <SignUpMessage />
      </MemoryRouter>,
    );

    const link = screen.getByRole('link');

    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute('href', ROUTES.REGISTER_PAGE);
  });
});
