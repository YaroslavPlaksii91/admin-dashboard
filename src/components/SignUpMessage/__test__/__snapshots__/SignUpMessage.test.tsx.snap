// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpMessage component should be rendered 1`] = `
<DocumentFragment>
  <div>
    <p
      class="MuiTypography-root MuiTypography-body1 css-c7xp52-MuiTypography-root"
    >
      Don't have an account?
    </p>
    <a
      href="/register"
    >
      Sign up
    </a>
  </div>
</DocumentFragment>
`;
